incubator/hnc/pkg/testutils/testutils.go

```diff
@@ -46,25 +47,49 @@ func RunShouldContain(substr string, seconds float64, cmdln ...string) {
 
 func RunShouldContainMultiple(substrs []string, seconds float64, cmdln ...string) {
 	Eventually(func() error {
-		stdout, err := RunCommand(cmdln...)
-		if err != nil {
+		return runShouldContainMultiple(substrs, cmdln...)
+	}, seconds).Should(Succeed())
+}
+
+func RunErrorShouldContain(substr string, seconds float64, cmdln ...string) {
+	RunErrorShouldContainMultiple([]string{substr}, seconds, cmdln...)
+}
+
+func RunErrorShouldContainMultiple(substrs []string, seconds float64, cmdln ...string) {
+	Eventually(func() error {
+		err := runShouldContainMultiple(substrs, cmdln...)
+		if containAll([]string {"Expecting: ", " but get: "}, err.Error()) {
 			return err
 		}
+		return nil
```

Comment:
It's generally considered to be bad practice to encode functional information in strings without a very well-defined schema (e.g., unless the string is JSON or something). One reason it that it's quite brittle - what happens if you change "but get" to "but got"? (The latter is actually more standard in Go). Another is that it's just confusing to the reader.

Instead, I'd suggest using multiple returned values in `runShouldContainMultiple`:

`found, err := runShouldContainMultiple(substrs, cmdln...)`

where `found` is true if everything was found, false otherwise, and `err` is only set if the command couldn't be run or returned a nonzero value. Or maybe:

`missing, err := runShouldContainMultiple(substrs, cmdln...)`

where `missing` contains any element from substrs that _wasn't_ found, so you can print them in a nice error message.

Reply:
done